### PR TITLE
1. Swap deprecated pedantic for lints

### PR DESCRIPTION
**Pull request by noga-dev**
2. Remove linter warnings by commenting out the code causing them
3. Add suggestion above the code that was causing linter warnings
3. Use final to encourage immutability

### CODE REVIEW

**Comment by noga-dev** [comment]
The dropping of pedantic in favor of lints is due to its deprecation per the description in the package page: https://pub.dev/packages/pedantic

**Comment by noga-dev** [comment]
5. Add analysis_options.yaml otherwise lint rules won't take effect

**Review by stnguyen90 — CHANGES_REQUESTED** [review]
Thanks for the PR! 🙏 Unfortunately, it looks like there are merge conflicts now. Would you please fix the merge conflicts?

**Review by stnguyen90 — APPROVED** [review]

File: dart-2.12/pubspec.yaml [hunk 1]
```diff
@@ -7,4 +7,4 @@ environment:
   sdk: '>=2.12.0 <3.0.0'
       
 dev_dependencies:
-  pedantic: ^1.9.0
\ No newline at end of file
+  lints: ^2.0.0
```
Comment (lohanidamodar) [inline dart-2.12/pubspec.yaml]: lints 2.0 requires Dart SDK 2.17 so this dart 2.12 starter shouldn't use lints 2.0.0

File: dart-2.13/pubspec.yaml [hunk 2]
```diff
@@ -7,4 +7,4 @@ environment:
   sdk: '>=2.12.0 <3.0.0'
       
 dev_dependencies:
-  pedantic: ^1.9.0
\ No newline at end of file
+  lints: ^2.0.0
```
Comment (lohanidamodar) [inline dart-2.13/pubspec.yaml]: same as above

**Review by lohanidamodar — CHANGES_REQUESTED** [review]
please fix the comment

**Review by lohanidamodar — DISMISSED** [review]
Looks good. Just that 2.0.1 is released let's update to that and get this merged.

**Comment by noga-dev** [comment]
> Looks good. Just that 2.0.1 is released let's update to that and get this merged.

2.0.1 of?

**Comment by christyjacob4** [comment]
@noga-dev `2.0.1` of the lints package has been released 😄  Please update to that 
https://pub.dev/packages/lints/versions

**Comment by noga-dev** [comment]
> @noga-dev `2.0.1` of the lints package has been released 😄 Please update to that https://pub.dev/packages/lints/versions

Done

**Review by christyjacob4 — APPROVED** [review]

**Comment by christyjacob4** [comment]
THANK YOU! All changes merged 🥳

Please reach out to me on our [Discord server](https://appwrite.io/discord) if you would like to claim your Appwrite swags! As a way of saying thank you, we would also love to invite you to join the Appwrite organization on GitHub. Please share your GitHub username with us on Discord.  

You can accept the invite by visiting https://github.com/orgs/appwrite/invitation. By joining our team, you will officially be an Appwrite maintainer on GitHub.

You can change your membership [visibility settings](https://help.github.com/en/github/setting-up-and-managing-your-github-user-account/publicizing-or-hiding-organization-membership), so your new Appwrite team membership badge will show up on your personal GitHub profile.

Please feel free to look for more PRs you might be interested in helping with on our long list of [Hacktoberfest friendly issues](https://github.com/search?q=org%3Aappwrite+org%3Autopia-php+org%3Aopen-runtimes+label%3Ahacktoberfest+created%3A%3E2021-01-01+state%3Aopen&type=issues&s=updated&o=asc) and help make Appwrite better :)

**Pull request merged**